subsys/bluetooth/controller/ll_sw/ull_sync_iso.c

```diff
@@ -11,29 +11,292 @@
 #include "common/log.h"
 #include "hal/debug.h"
 
+#include <sys/byteorder.h>
+
+#include "util/util.h"
+#include "util/mem.h"
+#include "util/memq.h"
+#include "util/mayfly.h"
+
+#include "hal/ccm.h"
+#include "hal/radio.h"
+#include "hal/ticker.h"
+
+#include "ticker/ticker.h"
+
+#include "pdu.h"
+#include "ll.h"
+
+#include "lll.h"
+#include "lll_vendor.h"
+#include "lll_clock.h"
+#include "lll_scan.h"
+#include "lll_sync.h"
+
+#include "ull_scan_types.h"
+#include "ull_sync_types.h"
+
+#include "ull_internal.h"
+#include "ull_scan_internal.h"
+#include "ull_sync_internal.h"
+#include "ull_sync_iso_internal.h"
+
+static struct ll_sync_iso ll_sync_iso_pool[CONFIG_BT_CTLR_SCAN_SYNC_ISO_MAX];
+static void *sync_iso_free;
+
+static int init_reset(void)
+{
+	/* Initialize sync pool. */
+	mem_init(ll_sync_iso_pool, sizeof(struct ll_sync_iso),
+		 sizeof(ll_sync_iso_pool) / sizeof(struct ll_sync_iso),
+		 &sync_iso_free);
+
+	return 0;
+}
+
+static inline struct ll_sync_iso *sync_iso_acquire(void)
+{
+	return mem_acquire(&sync_iso_free);
+}
+
+int ull_sync_iso_init(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int ull_sync_iso_reset(void)
```

Comment:
I assume this should be called from ull.c?

Reply:
Yup